### ISSUE TITLE
search results keywords not bold like traditonal search

### ISSUE THREAD

**Issue by cpluceDev** [microsoft-search/pnp-modern-search#481]
can we have the search keywords get bold in the results? when i use a search keyword in the search box it isnt bolded in the results

**Comment by Georg-Trixl** [comment]
I have it in solution for our company.
Use as column for the bold keyword **HitHighlightedSummary** and add that column in **Selected Properties**
and at the output side use following Handlebars Expression **{{getSummary HitHighlightedSummary}}**
Use this instead of **Description** (or what ever was used)
This should do the trick for you.

**Comment by cpluceDev** [comment]
Ah. So it's only applied in the description for highlighted summary. Won't work for other properties used if you customize result type. I'll test when I ge to the office later today.  Thanks

**Comment by cpluceDev** [comment]
Just to confirm. What you mean use as column highlighted summary? I need to name a column in my list this? So this would only apply to one column. Why if I had multiple detail properties in the result type? Is it possible to bold all fields or columns displayed in results

**Comment by cpluceDev** [comment]
thanks. i got this working but wonder what determines when to work or not. for instance, it worked in another list for other customs but it isnt working in another list for a custom column.

**Comment by Georg-Trixl** [comment]
Okay custom columns won't work I would say, I would also need to look into the source code (actually you can also do that on your own). But for it works for standard description fields and also content of text files like word or PDF.
But don't expect that columns containing content stored as number or as date will be highlighted.
If you are sure that customers need highlightening for 200% for other custom columns, then you could also write a custom script, for example with Handlebars. 
Or you use standard description field in your custom list and sell the description field as something else, by just changing it's title (should also work).

**Comment by cpluceDev** [comment]
i found this. https://docs.microsoft.com/en-us/sharepoint/search/how-to-display-values-from-custom-managed-properties-in-search-resultsoption-2

looks like we having indexing issues. closing issue. thx

**Comment by cpluceDev** [comment]
re open this because i think keywords in title in results should be bold as in classic search and isnt working. 

{{getSummary HitHighlightedSummary}} | Format the HitHighlightedSummary property with recognized words in bold.

this appears to work on some multiple line of text custom columns but not single line of text. si this by design?

**Comment by wobba** [comment]
Highlighting in the title is an improvement at best imo, as UX wise it doesn't look good. Check web search. Also if you want bold in other properties (which does make sense) that's an improvement and possibly new setting to specify hithighlighted properties.

**Comment by cpluceDev** [comment]
understood about bold in title. was comparing to the modern OOTB search where MSFT still does that but not a deal breaker.  there is not pattern but most other columns are getting bold and some are not. i think we are having an indexing issue as well which skewed my testing for this use case. can you elaborate on this? "and possibly new setting to specify hithighlighted properties."  What new setting? is this difficult? thx

**Comment by Georg-Trixl** [comment]
If my humble opinion and experience is of use ...
In our company we went for the cards layout - this means the Title is in big letters and trimmed in length (in case it's too large for the card), so the title can be read very fast. If there would be something in bold highlighted it would disturb in fast reading.
For description (usually for documents)  we provide more space and its written in smaller letters, the highlighting in the description is only a small excerpt of the document it self and shows that the keyword is contained and a little bit of the surrounding context but nothing more.
The highlight could lead to a successful search, but in the end showing the keyword in the surrounding does not show the quality of the search result and with that I mean the ranking of the result.
For example: 
1 Result contains the keyword 40 times 
2 Result contains the keyword only once 
Visually both sell as same high quality result - but when we start to think about that, then highlighting should be putted in question.
To personal the highlighting is only one small peace of the puzzle "how to get my users fast to the right result"

**Comment by cpluceDev** [comment]
i had a specific end user really request this feature in their search solution we are building.  so just wanted to confirm. this keyward bold in PnP modern search web part works similar or teh same as classic  EXCEPT the title words will not be strongtype. 

thx

**Comment by MakingThingsWorkToronto** [comment]
@cpluceDev are you able to submit a PR (new web component) to the MTW extensionsibilty library? We could do this as a pure UI thing pretty quickly - I won't have any time this week to do it unfortunately... A new component called Highlight with property for text & highlightKeyword? We can pass in both values...

**Comment by cpluceDev** [comment]
i can try. still learning the ropes here. but yea. this would be helpful and very much needed functionality. as the hithighledtsummary managed property is consistent. misses some columns in one list but works great in others. and some single line of text columns are missed. i dont see a pattern. so not sure why some columns dont get bolded and some do. same goes for OOTB classic search(although this shows more bold keywords) title isnt a necessity more than having consistent keywords get bolded across all column and not spot pick some but a nice add on thx

**Comment by cpluceDev** [comment]
so it appears to work on three lists. 1 doc lib and 2 custom lists. seems to work best with multiple line of text columns(custom or default). except for this one list where it isnt working with the multiple line of text column. not sure why this is. be nice to have it work with all column types

**Comment by cpluceDev** [comment]
update: still interested in this feature. seems i only get 3 columns form each of the 2 lists, total l6 that get the bold in the results. most are multiple and a couple are single line of text.  ideally, if i can get all columns to use this feature be great. ill try to a PR this week. reading up on how to do a PR. i'm sure this would be very similar to this approach but in spfx :
https://docs.microsoft.com/en-us/sharepoint/search/how-to-display-values-from-custom-managed-properties-in-search-resultsoption-2

**Comment by wobba** [comment]
@Georg-Trixl the hithighlighting oob is not super good with the current tech. A summary, hithighlighted or not takes up vertical space - and so far (based on AB experiments) it seems people have more success showing more results per page, compared to more information per result. But it's super hard to figure out if the extra information is not super crisp and helps you pick the correct item. If the summaries are super awesome, will less results in the viewport be better? :) 

That said - there are some options we can add to the webpart to make summaries better. I've POC'ed this myself, but the search index is not storing all needed values optimal today so unsure if I should add it or not (yet).

**Comment by cpluceDev** [comment]
i can close this as this isnt a need anymore. thanks
